### PR TITLE
(PA-3490) Add syslog logging to leatherman

### PR DESCRIPTION
**Pull request by GabrielNagy**

### CODE REVIEW

File: logging/inc/leatherman/logging/logging.hpp [hunk 1]
```diff
@@ -162,6 +162,13 @@ namespace leatherman { namespace logging {
      */
     void setup_eventlog_logging(std::string application);
 
+    /**
+     * Configures application to log to syslog
+     * The logging level is set to warning by default.
+     * @param application Destination stream for logging output.
+     */
+    void setup_syslog_logging(std::string application, int facility = (21<<3));
```
Comment (GabrielNagy) [inline logging/inc/leatherman/logging/logging.hpp:170]: should probably have a define for `(21<<3)`

Comment (ciprianbadescu) [reply logging/inc/leatherman/logging/logging.hpp:170]: I'm not sure if this parameter should be optional at all

File: logging/src/logging.cc [hunk 2]
```diff
@@ -168,6 +169,8 @@ namespace leatherman { namespace logging {
 
         if (use_event_log) {
             log_eventlog(level, message);
+        } else if (use_syslog) {
+            log_syslog(level, message);
         } else {
```
Comment (ciprianbadescu) [inline logging/src/logging.cc:174]: maybe we should replace use_event_log/use_event_log flags with logging_backend enum and make the dispatch using switch

Comment (ciprianbadescu) [reply logging/src/logging.cc:174]: we could also move boost related logging to `log_boost`

File: logging/inc/leatherman/logging/logging.hpp (hunk 1 above)
Comment (GabrielNagy) [reply logging/inc/leatherman/logging/logging.hpp:170]: makes sense

File: logging/src/posix/logging.cc [hunk 3]
```diff
@@ -39,4 +40,42 @@ namespace leatherman { namespace logging {
...
+        case log_level::error:
+            return LOG_ERR;
+        case log_level::warning:
+            return LOG_WARNING;
+        case log_level::info:
+            return LOG_INFO;
+        case log_level::debug:
+        case log_level::trace:
+        case log_level::none:
+            return LOG_DEBUG;
```
Comment (GabrielNagy) [inline logging/src/posix/logging.cc:69]: should not log anything if `level == log_level::none`

**Review by ciprianbadescu — APPROVED** [review]

**Pull request merged**